### ISSUE TITLE
Feature Request: Aerial Companion Updater

### ISSUE THREAD

**Issue by eric-schneider** [AerialScreensaver/Aerial#11]
Aerial Companion is _very_ good at updating [Aerial](https://github.com/JohnCoates/Aerial/), but not so good at updating itself! 😉 It would be great if Aerial Companion had options for automatically/manually checking for its own updates, and then install them just as easily as the Aerial screen saver updates.

**Comment by ZV00** [comment]
Hey @eric-schneider,

This is a suggestion that's been brought up a number of times. There are some complexities with it, but it's being seriously considered by @glouel.

**Comment by glouel** [comment]
@eric-schneider just to give a bit of feedback, when I started designing the companion, I thought people were going to hate the idea and wouldn't pick it up. And if they did they wouldn't want to update it. I've been super wrong on both counts ! 

So what's in right now, is a force update function that I never used. Basically, for security purpose, I can force you to download a new version of companion before being able to update Aerial. 

As I keep adding features though, I think I need to strongly reconsider everything and maybe use Sparkle for smooth updating.

**Comment by eric-schneider** [comment]
@glouel Well, I think the app is super helpful and a pleasure to use. No wonder it's so popular!

I think the force update function you describe should work in the near-term to accommodate major changes in Aerial or if vulnerabilities are discovered in Aerial Companion.

But yes, if more and more useful enhancements are made to Aerial Companion, it would make sense to add Sparkle. Having Sparkle up and running before the first Apple Silicon Macs are GA could be handy for squashing any bugs that pop up on real-world hardware.

**Comment by glouel** [comment]
@eric-schneider thanks ! Based on feedback I'm adding more and more features (though it's not released yet) like running in a window/wallpaper and things of the sort. 

<img width="485" alt="Capture d’écran 2020-08-30 à 12 49 19" src="https://user-images.githubusercontent.com/37544189/91747193-0c3f2580-ebbe-11ea-9217-e2e6c3c218ad.png">

Still have a few bugs to iron out there as I had to go back to objective C to get that working (loading the installed screen saver inside companion). 

So yeah Sparkle may be a great solution, that and your point on Apple Silicon being GA being spot on (it does work great on DTK for now which is nice, if anything Apple Silicon is *great* for Aerial with perfect hardware video decoding ;))

**Comment by dVIAN13** [comment]
Hey @glouel! I wanted to bump this thread to see if there were any updates? pun intended 🤣

**Comment by glouel** [comment]
Hey @dVIAN13 ! Still not but I finally started working again on it today, releasing a new build. It does fix a few issues but I still haven't added the self update thing, but I'll try to get to it soon 🙈

**Comment by dVIAN13** [comment]
Thanks @glouel ! As always, amazing progress!

**Comment by Pcar17** [comment]
> Thanks @glouel ! As always, amazing progress!

Yes, absolutely!

**Comment by glouel** [comment]
Yeah I don't know about that, I've been pretty slow working on Companion those last few months 😅 But thanks for the kind words !

**Comment by glouel** [comment]
V4 auto updates everything via Sparkle again 😅
